contents/stats_vis.md

```diff
@@ -1,7 +1,181 @@
+It shows us the full distribution of a given variable by depicting it as a density, where the higher the curve at a given point more likely is the variable to take certain value.
+
+The density plot can also be drawn using `Makie.jl`, however it is more convoluted than the box plot.
+First, we want to pass for each `density!` function only the values with respect to one observation.
+Thus, we define a `values` function that will accept a `code` argument to filter the dataset's variable `name` wrapped with the `levelcode` function.
+Then, we plot a density `pltobj` for each one of the variable `name`'s `levels`.
+Finally, we make sure that the density `plotobj`s have their own `ytick` with the `offset` keyword paired with a custom `yticks` in the `Axis` constructor by specifying, same as before, a tuple of values and labels.
+The effect of the `offset` in the `for` loop is the increment from 1 to 4, by 1, of both the `offset` argument for `density!` and the `code` argument for `values`:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "densityplot" # hide
+    caption = "Density Plot" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    categories = levels(df.name)
+    values(code) = filter(row -> levelcode.(row.name) == code, df).grade
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; yticks = (1:4, categories), limits=((-1, 11), nothing))
+    for i in 1:length(categories)
+        density!(ax, values(i); offset=i)
+    end
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+As explained in @sec:makie_colors, we can change Makie's colors by either specifying a `color` or ` colormap`.
+This can also be applied to `density`:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "densityplot_colors" # hide
+    caption = "Density Plots with Different Color Schemes" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    categories = levels(df.name)
+    values(code) = filter(row -> levelcode.(row.name) == code, df).grade
+    fig = Figure(; resolution=(600, 400))
+    ax1 = Axis(fig[1, 1]; yticks = (1:4, categories), limits=((-1, 11), nothing))
+    ax2 = Axis(fig[1, 2]; yticks = (1:4, categories), limits=((-1, 11), nothing))
+    for i in 1:length(categories)
+        density!(ax1, values(i); offset=i, color=(:dodgerblue, 0.5))
+    end
+    for i in 1:length(categories)
+        density!(ax2, values(i); offset=i, color=:x, colormap=:viridis)
+    end
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+Here, in the first figure (left) we are using a specific `color` for all `density!`'s `plotobj`s.
+And in the second figure (right) we pass the `:x` argument to `color` to tell Makie to apply the `colormap` gradient along the x-axis (from left to right) while also specifying which `colormap` palette as `:viridis`.
```

Comment:
Could you also specify the benefit of this?

Reply:
😂 I don't have any. This is just customizations for the user to do as pleased.

Reply:
The color code gradient in the `y` direction is most common and is a visual aid to easily identify trends. In the `x` direction is useful when you want to know how things go in time, distance, etc... (your variable), but is not widely used.

Reply:
This is good I will add this /\